Userland/Services/WindowServer/WindowManager.cpp

```diff
@@ -1965,11 +2032,11 @@ ResizeDirection WindowManager::resize_direction_of_window(Window const& window)
     return m_resize_direction;
 }
 
-Gfx::IntRect WindowManager::tiled_window_rect(Window const& window, WindowTileType tile_type, bool relative_to_window_screen) const
+Gfx::IntRect WindowManager::tiled_window_rect(Window const& window, Screen const* cursor_screen, WindowTileType tile_type) const
 {
     VERIFY(tile_type != WindowTileType::None);
 
-    auto& screen = Screen::closest_to_rect(window.frame().rect());
+    auto const& screen = cursor_screen ? *cursor_screen : Screen::closest_to_rect(window.frame().rect());
```

Comment:
Nit: Maybe an `Optional<Screen&>` is a nicer type? Then this could be `cursor_screen.value_or()`.

Reply:
I don't mind using `Optional<Screen&>` but let's keep in mind that using `value_or()` will cause the potentially a lot of wasted effort determining the alternate value, whether it's needed or not. Not a huge overhead in this case, but we should probably use this only in cases where we already have an alternate value available regardless of the condition.